### ISSUE TITLE
sitemap integration removes pages ending with 404 or 500

### ISSUE THREAD

**Issue by adrianlyjak** [withastro/astro#10778]
### Astro Info

```block
Astro                    v4.2.3
Node                     v20.11.1
System                   macOS (arm64)
Package Manager          npm
Output                   hybrid
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/react
                         @astrojs/sitemap
```

### Describe the Bug

I am using numerical IDs for items on a static site. One of the items (a store location) has an ID of "404". Its getting stripped from the sitemap, since 404 pages are a somewhat special identifier to be used as a not found page in some cases for astro.

It seems that perhaps the exclusionary logic is a little too strict. Can this be made to instead only match this type of pattern?

```
/404
/500
/[locale]/404
/[locale]/500
```

In my case the page follows this pattern

```
/store/[state]/[city]/[store-id]
```

### What's the expected result?

Not all pages that end in 404 should be excluded from the site map

### Link to Minimal Reproducible Example

https://github.com/adrianlyjak/astro/tree/404-fix/packages/integrations/sitemap/test/fixtures/static

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
@ascorbic I think this one shouldn't be too hard, suspect the logic here is too strict: https://github.com/withastro/astro/blob/main/packages/integrations/sitemap/src/index.ts#L50

**Comment by adrianlyjak** [comment]
@matthewp I think this can actually be closed? I had an PR (#10779) that was merged last month. The updated sitemap plugin is working correctly for my use case now

**Comment by ematipico** [comment]
@adrianlyjak you can close it :)
